plugins/woocommerce/tests/php/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -1581,4 +1581,268 @@ public function test_collection_hand_picked_order() {
 		$this->assertEquals( $product_ids, $result_frontend['post__in'] );
 		$this->assertEquals( $product_ids, $result_editor['post__in'] );
 	}
+
+	/**
+	 * Tests that the by-category collection handler works as expected.
+	 */
+	public function test_collection_by_category() {
```

Comment:
Just a comment: this file is starting to be huge. We need to split tests. 

Did you have a chance to explore [this](https://github.com/woocommerce/woocommerce/pull/57892#issuecomment-2886666468)?